src/serac/numerics/solver_config.hpp

```diff
@@ -100,31 +100,64 @@ struct TimesteppingOptions {
 enum class LinearSolver
 {
-  CG,       /**< Conjugate gradient */
-  GMRES,    /**< Generalized minimal residual method */
-  SuperLU,  /**< SuperLU MPI-enabled direct nodal solver */
-  Strumpack /**< Strumpack MPI-enabled direct frontal solver*/
+  CG,        /**< Conjugate gradient */
+  GMRES,     /**< Generalized minimal residual method */
+  SuperLU,   /**< SuperLU MPI-enabled direct nodal solver */
+  Strumpack, /**< Strumpack MPI-enabled direct frontal solver*/
+  PetscCG,   /**< PETSc MPI-enabled conjugate gradient solver */
+  PetscGMRES /**< PETSc MPI-enabled generalize minimal residual solver */
 };
 // _linear_solvers_end
 
+/// Convert linear solver enums to their string names
+inline std::string linearName(const LinearSolver& s)
+{
+  if (s == LinearSolver::CG) return "CG";
+  if (s == LinearSolver::GMRES) return "GMRES";
+  if (s == LinearSolver::SuperLU) return "SuperLU";
+  if (s == LinearSolver::Strumpack) return "Strumpack";
+  if (s == LinearSolver::PetscCG) return "PetscCG";
+  if (s == LinearSolver::PetscGMRES) return "PetscGMRES";
+  return "";
+}
+
 // Add a custom list of strings? conduit node?
 // Arbitrary string (e.g. json) to define parameters?
 
 // _nonlinear_solvers_start
 /// Nonlinear solver method indicator
 enum class NonlinearSolver
 {
-  Newton,           /**< MFEM-native Newton-Raphson */
-  LBFGS,            /**< MFEM-native Limited memory BFGS */
-  NewtonLineSearch, /**< Custom solver using preconditioned earch direction with backtracking line search */
-  Nesterov,         /**< Custom solver using Nesterov dynamic (damped) dynamics for accelerating implicit solves */
-  TrustRegion,      /**< Custom solver using a trust region solver */
-  KINFullStep,      /**< KINSOL Full Newton (Sundials must be enabled) */
+  Newton,                    /**< MFEM-native Newton-Raphson */
+  LBFGS,                     /**< MFEM-native Limited memory BFGS */
+  NewtonLineSearch,          /**< Custom solver using preconditioned earch direction with backtracking line search */
+  TrustRegion,               /**< Custom solver using a trust region solver */
+  KINFullStep,               /**< KINSOL Full Newton (Sundials must be enabled) */
   KINBacktrackingLineSearch, /**< KINSOL Newton with Backtracking Line Search (Sundials must be enabled) */
-  KINPicard                  /**< KINSOL Picard (Sundials must be enabled) */
+  KINPicard,                 /**< KINSOL Picard (Sundials must be enabled) */
+  PetscNewton,               /**< PETSc Full Newton */
+  PetscNewtonBacktracking,   /**< PETSc Newton with backtracking line search */
+  PetscNewtonCriticalPoint,  /**< PETSc Newton with critical point line search */
+  PetscTrustRegion           /**< PETSc trust region solver */
 };
 // _nonlinear_solvers_end
 
+/// Convert nonlinear linear solver enums to their string names
+inline std::string nonlinearName(const NonlinearSolver& s)
+{
+  if (s == NonlinearSolver::Newton) return "Newton";
```

Comment:
Case statement.

Should this error or is it caught by the calling function?

Reply:
We don't need the empty string case if we use a switch statement

Reply:
Just kidding, GCC isn't smart enough to figure that out. I'm setting it to "UNKNOWN"